docs/react-components-structure.md

```diff
@@ -0,0 +1,161 @@
+  splitted in two: the pages entry point and the modules which can be reused in
+  multiple pages.
+
+In order to understand why, let's go through the history of Front-Commerce.
+
+## Step 1: Atomic Design
+
+Back in early 2016, we started with the will to apply the
+[Atomic Design principles](http://bradfrost.com/blog/post/atomic-web-design/).
+Indeed, it is a really good canvas for those who want to put their feet in the
+Design System world. It gave us guidelines that should help to keep our theme
+nice, clean and consistent. We only needed to split our components in 5 folders
+: atoms, molecules, organisms, templates and pages.
+
+http://bradfrost.com/wp-content/uploads/2013/06/atomic-design.png
+
+And overall, it seemed like a perfect fit with React which embraced the ideas of
+components. We believed that by splitting our components in tiny little pieces
+that would assemble together to create whole pages, it would be easy to apply
+those principles. Thus, we ended with this kind of structure:
+
+```
+src/
+├── ...
+└── web/
+    └── theme/
+        ├── atoms/
+        ├── molecules/
+        ├── organisms/
+        ├── templates/
+        └── pages/
+```
+
+## Step 2: Fuse templates and pages
+
+And we built our first components eagerly! Everything went well until we built
+our first templates/pages components. Indeed, since there was no distinction
+between UI and business logic, templates were already full grown pages.
+
+So we just considered that the difference didn't stand in our case and got rid
+of `pages` folder and move the components to the `templates` folder.
+
+```
+src/
+├── ...
+└── web/
+    └── theme/
+        ├── atoms/
+        ├── molecules/
+        ├── organisms/
+        └── templates/
+```
+
+## Step 3: Split business and UI logic
+
+The core product continued to grow, we had more and more components and the sun
+was shiny. But after a while we felt that the theme was inconsistent. Even
+though it was the whole point of Atomic Design, we failed at building a
+homogeneous theme.
+
+So we stopped and looked back for a bit. What was wrong here?
+
+The root cause was that it was really hard to find existing components. It
+leaded to many dupplicates. The theme components were actually drowned in the
+middle of the business logic components.
+
+That's why we brought back the `pages` components back. They would be the React
+Components containing specific business logic, fetching data from the outside
+and giving life to UI components. In order to emphasize this decision, we
+created a new folder called `components` which would contain all the UI
+components (`atoms`, `molecules`, `organisms` and `templates`).
+
+```
+src/
+├── ...
+└── web/
+    └── theme/
+        ├── components/
+        |   ├── atoms/
+        |   ├── molecules/
+        |   ├── organisms/
+        |   └── templates/
+        └── pages/
+```
+
+And everything seemed pretty perfect here. The theme took back its consistency
```

Comment:
I don't really like using "pretty" with "perfect" here, perfect being an absolute notion.

It is a bit nitpicky though, I can't really think of a better alternative except simply removing "pretty"